### DISCUSSION TITLE
TS2428 error when using custom middleware that modifies the store type and persist middleware

### DISCUSSION THREAD

**Discussion by pharaok** [pmndrs/zustand#1851] [Bug report] [answered]
Hi, I have a next app where I'm using the persist middleware and my own middleware that modifies the store type.

I'm just following the example from the typescript guide and as soon as I import the persist middleware somewhere else in the app, it says "All declarations of 'StoreMutators' must have identical type parameters.".

here's a [CodeSandbox reproduction](https://codesandbox.io/s/condescending-tesla-48tn3k?file=/src/index.tsx:163-211), as soon as you uncomment the line importing the persist middleware, the StoreMutators declaration breaks.

**Comment by dai-shi** [comment, marked as answer]
I don't know exactly, but is it the order of import? https://codesandbox.io/s/misty-waterfall-fpwq8m

**Reply by pharaok** [reply]
For some reason that fixes it, but to use this workaround in my project I have to import the persist middleware in my middleware file even though it's not used there.

I guess that can also explain why changing the file name seemed to produce different results, maybe the bundler sorted the imports differently based on that? Pretty weird either way though.

Thank you!

**Reply by dai-shi** [reply]
I can't explain. We need a TS wizard to explain it.
